src/crepe/api/UiObject.h

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include "../Component.h"
+
+namespace crepe {
+
+/**
+ * @class UiObject
+ * \brief Represents a UI object in the game, derived from the Component class.
+ */
+class UiObject : public Component {
+public:
+	/**
+     * \brief Constructs a UiObject with the specified game object ID.
+     * \param id The unique ID of the game object associated with this UI object.
+     */
+	UiObject(game_object_id_t id);
+
+	//! The width of the UI object.
+	int width = 0;
+
+	//! The height of the UI object.
+	int height = 0;
```

Comment:
Even voor de zekerheid: width en height zijn in game units (niet in pixels), toch?

Reply:
ja deze data moet niels implementeren in het render system